src/foam/mlang/mlangJava.js

```diff
@@ -323,6 +323,20 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  refines: 'foam.mlang.predicate.Has',
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: 'Object value = getArg1().f(obj); '+
+        'return ! (value == null || '+
```

Comment:
yes, I commented on this previously, but it was lost/hidden:
"Testing for an empty array is not straight forward. Testing for an array can be accomplished with 'value.getClass().isArray()' or 'instanceof Type[]', but then to get the length, it needs to be cast or you have to know the Type."
How would you like proceed?